File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
import java.util.*;

class Solution {
    static HashSet<Integer> set = new HashSet<Integer>();
    static boolean[] arr = new boolean[1000001];
    static String check = "";
    static Stack<Integer> q = new Stack<>();
    static int ans = 0;

    public static int solution(int n, int k) {
        bit(n, k);

        int answer = ans;
        return answer;
    }

    public static void bit(int bit, int k) {
        while (bit != 0) {
            if (bit % k == 0) {
                String s = "";
                if(q.isEmpty()){

                }
                else{
                    while (!q.isEmpty()) {
                        s += q.pop();
                    }
                    if (Function(Long.parseLong(s))) {
                        ans++;
                    }
                }
            }
            else {
                q.add(bit % k);
            }
            bit /= k;
        }
        String s = "";
        if(q.isEmpty()){

        }
        else{
            while (!q.isEmpty()) {
                s += q.pop();
            }
            if (Function(Long.parseLong(s))) {
                ans++;
            }
        }
    }

    public static boolean Function(long num)
    {
        if (num < 2)
            return false;

        for (int i = 2; i <= Math.sqrt(num); i++)
            if (num % i == 0)
                return false;

        return true;
    }
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
import math
def is_prime(num):
    if num<2:
        return False
    for i in range(2,int(math.sqrt(num))+1):
        if num % i ==0:
            return False
    return True
    
def solution(n, k):
    count = ""
    answer = 0
    while n != 0:
        count= str(n%k)  + count
        n//=k
    list1 = list(count.split("0"))
    for i in range(len(list1)):
        if list1[i] != "" and is_prime(int(list1[i])):
            answer +=1
    return answer